site/plugins.md

```diff
@@ -390,8 +391,254 @@ func (p *Plugin) Report(w http.ResponseWriter, r *http.Request) {
 
 ```
 
-**See Also**
+### <a id="report-data-structures"></a>Report Data structures
+A report can contain many types of information.
+If you go back to the [Reporter Interface](#reporter-interface) section, you will see the top-level `Plugins` attribute.
+Along with that, a report may contain multiple topologies.
+An example of a report containing a few topologies is the following:
 
-  * [Building Scope](/site/building.md)
+```json
+{
+  "Host": {},
+  "Container": {},
+  "Process": {},
+  ...,
+  "Plugins": [...,]
+}
+```
+
+### Topologies
+Topologies can be of various types.
+Each topology consists of some general information and a list of nodes.
+These are the available topologies:
+
+- `Endpoint` nodes are `(address, port)` tuples on each host.
+- `Process` nodes are processes on each host.
+- `Container` nodes represent all Docker containers on hosts running probes.
+- `Pods` nodes represent all Kubernetes pods running on hosts running probes.
+- `Service` nodes represent all Kubernetes services running on hosts running probes.
+- `Deployment` nodes represent all Kubernetes deployments running on hosts running probes.
+- `ReplicaSet` nodes represent all Kubernetes ReplicaSets running on hosts running probes.
```

Comment:
we now also have `ECSTask` and `ECSService` representing ECS Tasks and Services respectively.